examples/go-client/main.go

```diff
@@ -18,26 +18,62 @@ package main
 
 import (
 	"context"
-	"fmt"
 	"log"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+
+	"github.com/gravitational/trace"
+	"github.com/pborman/uuid"
 )
 
 func main() {
+	ctx := context.Background()
 	log.Printf("Starting Teleport client...")
-	client, err := connectClient()
+
+	creds, err := client.ProfileCreds()
+	if err != nil {
+		log.Fatalf("Failed to read credentials: %v", err)
+	}
+
+	clt, err := client.New(client.Config{
+		Addrs: []string{"proxy.example.com:3025"},
+		Creds: creds,
+	})
 	if err != nil {
 		log.Fatalf("Failed to create client: %v", err)
 	}
-	defer client.Close()
+	defer clt.Close()
 
-	ctx := context.Background()
+	demoClient(ctx, clt)
+}
 
-	fmt.Println("")
-	roleCRUD(ctx, client)
+func demoClient(ctx context.Context, clt *client.Client) (err error) {
+	// Create a new access request for api-admin to use the admin role.
+	accessReq, err := types.NewAccessRequest(uuid.New(), "access-admin", "admin")
+	if err != nil {
+		return trace.Wrap(err, "Failed to make new access request: %v")
+	}
+	if err = clt.CreateAccessRequest(ctx, accessReq); err != nil {
+		return trace.Wrap(err, "Failed to create access request: %v")
+	}
+	log.Printf("Created access request: %v", accessReq)
+
+	defer func() {
+		if err = clt.DeleteAccessRequest(ctx, accessReq.GetName()); err != nil {
+			log.Fatalf("Failed to delete access request: %v", err)
```

Comment:
I would not use `FatalX` API here since the function is supposed to return an error. Since it is already affecting the error value it should be fine as is. One thing to note though is that since this `defer` runs regardless of the current value of err, it will overwrite a legitimate error from the main function body, so I'd take that into account even though it's just an example code.